auction-house/program/src/lib.rs

```diff
@@ -965,6 +966,89 @@ pub mod auction_house {
         Ok(())
     }
 
+    pub fn print_receipt<'info>(
+        ctx: Context<'_, '_, '_, 'info, PrintReceipt<'info>>,
+        trade_state_bump: u8,
+        receipt_bump: u8,
+        price: u64,
+        token_size: u64,
+    ) -> ProgramResult {
+        let trade_state = &ctx.accounts.trade_state;
+        let receipt = &mut ctx.accounts.receipt;
+        let auction_house = &ctx.accounts.auction_house;
+        let token_account = &ctx.accounts.token_account;
+        let bookkeeper = &ctx.accounts.bookkeeper;
+        let wallet = &ctx.accounts.wallet;
+
+        receipt.trade_state = trade_state.key();
+        receipt.auction_house = auction_house.key();
+        receipt.token_account = token_account.key();
```

Comment:
Please audit me but not saving the metadata address because it can be looked up based on the mint of the token account.